src/queries/labelText.ts

```diff
@@ -17,12 +18,48 @@ type ByLabelTextOptions = CommonQueryOptions & TextMatchOptions;
 const getNodeByLabelText = (
   node: ReactTestInstance,
   text: TextMatch,
-  options: TextMatchOptions = {}
+  options: TextMatchOptions
 ) => {
   const { exact, normalizer } = options;
   return matches(text, node.props.accessibilityLabel, normalizer, exact);
 };
 
+const matchAccessibilityLabelledByText = (
+  rootInstance: ReactTestInstance,
+  nativeID: string | undefined,
+  text: TextMatch,
+  options: TextMatchOptions
+) => {
+  if (!nativeID) {
+    return false;
+  }
+
+  const { exact, normalizer } = options;
+
+  return rootInstance
+    .findAll((node) =>
+      matches(nativeID, node.props?.nativeID, normalizer, exact)
```

Comment:
`React.Fragment` doesn't have neither `nativeID` prop nor `testID` and tests started to fail. I've extended the type of `text` in `matches` to `string | undefined`, because we are checking it anyway:
```
 if (typeof text !== 'string') {
    return false;
  }
```
Adding `typeof node.type === 'string'` would fix it, but then we have to add 
```
const isTextComponent = filterNodeByType(node, Text) | filterNodeByType(node, "Text")
```
to `matchTextContent`. 
I didn't know what fits best here. @mdjastrzebski